Service creation

AtaKaleli opened the pull request.

AtaKaleli committed:
Movie Interface, Repository and Controller is added

AtaKaleli committed:
Theatre Interface,Repository and Controller added

AtaKaleli committed:
Worker Interface, repository and controller added

AtaKaleli committed:
Scopes are added to main

moh-nabih commented on WebApplication1/Repositories/MovieRepository.cs:
```diff
@@ -0,0 +1,19 @@
+        {
+            _context = context;
+        }
+        public ICollection<Movie> GetMovies()
```
this should return a List specifically not ICollection, since this is the data you are going to use

moh-nabih replied:
same for the others

AtaKaleli replied:
sir, in the repositories, I returned  ().ToList(); Get method. Do I need to change it anyway?

moh-nabih commented on WebApplication1/Controllers/MovieController.cs:
```diff
@@ -0,0 +1,28 @@
+            if (!ModelState.IsValid)
+                return BadRequest(ModelState);
+
+            return Ok(movies);
```
you should use separate models to use for each layer, so for example, if you have Movie, you should have MovieModel for service layer and MovieViewModel for controller layer

AtaKaleli committed:
Icollections are changed to list in interface and repository files

AtaKaleli committed:
Controllers and Repositories are updated

To ensure we have a connection between movie and director, I added some extra codes in repositories

AtaKaleli committed:
Get with Id endpoint added

AtaKaleli committed:
Get with endpoint added to theatre

AtaKaleli committed:
get by ID endpoint added to worker

moh-nabih commented on WebApplication1/Controllers/WorkerController.cs:
```diff
@@ -0,0 +1,46 @@
+            return Ok(workers);
+        }
+        [HttpGet("{id}")]
+        [ProducesResponseType(200, Type = typeof(Worker))]
```
no need for these, also leave a space between methods

moh-nabih replied:
do that for other controllers too, we don't need the produces response types

AtaKaleli committed:
ProducesResponseTypes are deleted

moh-nabih commented on WebApplication1/Controllers/TheatreController.cs:
```diff
@@ -0,0 +1,30 @@
+
+        [HttpGet]
+        [ProducesResponseType(200, Type = typeof(IEnumerable<Theatre>))]
+        public IActionResult GetTheatres()
```
```suggestion
        public IActionResult<List<Theatre>> GetTheatres()
```

moh-nabih replied:
do this for the rest
